packages/framework/client-logger/external-telemetry/src/container/telemetryManager.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IContainer, type ICriticalContainerError } from "@fluidframework/container-definitions";
+import { IContainerTelemetry } from "./containerTelemetry";
+import { ContainerEventTelemetryProducer } from "./telemetryProducer";
+import { ITelemetryConsumer } from "../common";
+import { ContainerSystemEventName, ContainerSystemEventNames } from "./containerSystemEvents";
+
+/**
+ * This class manages container telemetry intended for customers to consume.
+ * It manages subcribing to the proper raw container system events, sending them to the {@link ContainerEventTelemetryProducer}
+ * to be transformed into {@link IContainerTelemetry} and finally sending them to the provided {@link ITelemetryConsumer}
+ *
+ * @internal
+ */
+export class ContainerTelemetryManager {
+	constructor(
+		private readonly container: IContainer,
+		private readonly telemetryProducer: ContainerEventTelemetryProducer,
+		private readonly telemetryConsumer: ITelemetryConsumer,
+	) {
+		this.setupEventHandlers(this.container);
+	}
+
+	/**
+	 * Subscribes to the raw container system events and routes them to telemetry producers.
+	 */
+	private setupEventHandlers(container: IContainer) {
```

Comment:
Nit: no big advantage in passing the container as a parameter if it's already available anywhere in the class, we can just do `this.container` in the lines below. If the function was reusable across different containers then it'd make sense, but in this case I'd say let's remove it to avoid any potential confusion.

Reply:
Sure. Updated.